src/components/AddPlaidBankAccount.js

```diff
@@ -129,29 +91,26 @@ class AddPlaidBankAccount extends React.Component {
             return;
         }
 
-        BankAccounts.clearPlaidBankAccountsAndToken();
-        BankAccounts.fetchPlaidLinkToken();
-    }
-
-    componentWillUnmount() {
-        BankAccounts.setBankAccountFormValidationErrors({});
+        BankAccounts.clearPlaid();
+        BankAccounts.openPlaidBankLogin(this.props.allowDebit, this.props.bankAccountID);
     }
 
     /**
      * Get list of bank accounts
      *
      * @returns {Object[]}
      */
-    getAccounts() {
-        return lodashGet(this.props.plaidBankAccounts, 'accounts', []);
+    getPlaidBankAccounts() {
+        return lodashGet(this.props.plaidData, 'bankAccounts', []);
     }
 
     /**
      * @returns {String}
      */
     getPlaidLinkToken() {
-        if (!_.isEmpty(this.props.plaidLinkToken)) {
-            return this.props.plaidLinkToken;
+        const plaidLinkToken = this.props.plaidLinkToken;
```

Comment:
NAB: not sure this really needs to be its own variable since it's just `this.props.plaidLinkToken`. Why not:

```JS
if (this.props.plaidLinkToken) {
     return this.props.plaidLinkToken;
}
```